keanu-project/src/main/java/io/improbable/keanu/vertices/LogProbGraph.java

```diff
@@ -0,0 +1,34 @@
+package io.improbable.keanu.vertices;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+
+public class LogProbGraph {
+
+    private final Map<Vertex<?>, Vertex<?>> inputs;
```

Comment:
A graph without random variables or probabilistic would still have `vertices` right? A Bayesian Network is just a special case of a graph and a graph is composed of vertices. I think we need to rename some of our interfaces so that we either have `RandomVariables<T>` or `Operation<T>` which will both be a `Vertex<T>`. In that case this mapping on inputs could still be either from RandomVariable to Operation or from Operation to Operation.

Reply:
Agreed - let's attack the problem in terms of interfaces.
* What interface is only for BayesNet Vertices? `sample`, `observe` etc.
* What interface is common to both types? `eval`, `addChild` etc.

Reply:
I was suggesting that we refactor the way we've currently set interfaces on our Vertex classes. There should be two types of vertices, RandomVariables and Operation with the only difference being the way we allow the user to observe them. Currently we've defined them as Probabilistic and NonProbabilistic but I think that's wrong. For the purposes of this mapping, in both cases it's a mapping of vertex to vertex. The tightest you could bound it would be Map<Vertex<?>, Operation<?>> but I don't think the tighter bounds gives you anything except added complexity.